buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -744,8 +744,11 @@ private void deallocate() {
                 allocatedBytes = 0;
                 if (!mag.trySetNextInLine(this)) {
```

Comment:
What we could also do is just create a new `Chunk` object that wraps the delegate. This way we would not need to call `resetRefCnt()` and so would not need to do the release dance below. @chrisvest WDYT ?

Reply:
Yeah it produce a bit more GC but maybe it's safer

Reply:
https://github.com/netty/netty/pull/14393